velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -786,19 +799,65 @@ TypePtr ReaderBase::convertType(
             schemaElement.type,
             thrift::Type::INT64,
             "TIMESTAMP_MICROS or TIMESTAMP_MILLIS converted type can only be set for value of thrift::Type::INT64");
+        VELOX_CHECK(
+            !requestedType || requestedType->kind() == TypeKind::TIMESTAMP,
+            kTypeMappingErrorFmtStr,
+            "TIMESTAMP",
+            requestedType->toString());
         return TIMESTAMP();
 
       case thrift::ConvertedType::DECIMAL: {
         VELOX_CHECK(
             schemaElement.__isset.precision && schemaElement.__isset.scale,
             "DECIMAL requires a length and scale specifier!");
-        return DECIMAL(schemaElement.precision, schemaElement.scale);
+        const auto schemaElementPrecision = schemaElement.precision;
+        const auto schemaElementScale = schemaElement.scale;
+        // A long decimal requested type cannot read a value of a short decimal.
+        // As a result, the mapping from short to long decimal is currently
+        // restricted.
+        auto type = DECIMAL(schemaElementPrecision, schemaElementScale);
+        if (requestedType) {
+          VELOX_CHECK(
+              requestedType->isDecimal(),
+              kTypeMappingErrorFmtStr,
+              "DECIMAL",
+              requestedType->toString());
+          // Reading short decimals with a long decimal requested type is not
+          // yet possible. To allow for correct interpretation of the values,
```

Comment:
scale of the requestedType can be larger than the file column.
We only need to ensure the physical type byte counts are compatible.

Reply:
We discussed this offline. The implementation ensures file and requested decimal types match.